docs/userGuide/syntax/questions.mbdf

```diff
@@ -1,130 +1,325 @@
 <variable name="highlightStyle">html</variable>
 <variable name="code">
-<question>
-  This question with an omitted hint.
-  <div slot="answer">42 <md>:star:</md></div>
-</question>
 
-<question>
-  This question with an omitted answer.
-  <div slot="hint">The number of exponent bits in a 64-bit Floating Point. <md>:computer:</md></div>
-</question>
+<question type="checkbox">
 
-<question>
-  This question only has the question body.
+##### Which of the following is true?<br>
+<q-option reason="lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum">
+  1 + 1 = 11
+</q-option>
+<q-option reason="Division by zero is **undefined**!">
+  10 / 0 = 10
+</q-option>
+<q-option correct>
+11 / 11 = 1
+</q-option>
+<template slot="hint">
+
+Use your calculator! :fas-calculator:
+</template>
 </question>
+
 </variable>
 </include>
 
-**Use the minimized Panel component to create and group questions inline.**
+****`q-option` Options and Slots****
+Name | Type | Default | Description
+--- | --- | --- | ---
+correct | `Boolean` | `false` | Whether the answer (checkbox or mcq) is correct. You may have multiple correct answers in either case.
+reason  | `String` | `''` | The explanation to display for the option once the answer is checked.
+reason  | Slot | empty | The explanation to display for the option once the answer is checked. Overrides the reason option if both are present.
 
-<include src="codeAndOutput.md" boilerplate >
-<variable name="highlightStyle">html</variable>
-<variable name="code">
-<panel header="Q1 :crown:" minimized>
-  <question has-input>
-    Who is the first king of Thailand?
-  </question>
-</panel>
-<panel header="Q2 :pizza:" minimized>
-  <question has-input>
-    Which country did the Hawaiian pizza originate from?
-    <div slot="hint">
-      Not Italy or Haiwaii <md>:smirk:</md>
-    </div>
-  </question>
-</panel>
-</variable>
-</include>
 
-**Questions can also be loaded in from another file using Panel's `src` attribute.**
+#### Text Questions
 
-<include src="codeAndOutput.md" boilerplate >
+Text questions are specified with the `type="text"` attribute.
+
+Unlike mcq and checkbox questions, answer checking is performed by providing keywords to check for in the user's answer through the `keywords` attribute.
+If no keywords are provided, the answer will always be marked as correct.
+
+Similarly, you can provide your answer / explanation in the `answer` attribute or `<template slot="answer">` slot.
+
+<include src="codeAndOutput.md" boilerplate>
 <variable name="highlightStyle">html</variable>
 <variable name="code">
-<panel header="Questions loaded in from `quiz.md`" src="extra/quiz.md"></panel>
+<question type="text" header="Which country did the Hawaiian pizza originate from?"
+          keywords="hawaii" fuzzy="1" threshold="0.5" answer="It originated from Hawaii!">
+  <template slot="hint">
+
+  Watch some pizza commercials!
+
+  :pizza: :pizza: :pizza: :pizza: :pizza: :pizza: :pizza:
+
+  </template>
+</question>
```

Comment:
Any mention of whether _all_ keywords need to match?

Reply:
added a mention, there's a `threshold` option to configure it as well

Reply:
I forgot to mention the checking algorithm, currently it just does case normalization + a naive substring search, so if you have a keyword `xx`, `yxxy` would also match.

Not sure if it should be more restrictive:
1. splitting by whitespace + case normalization
2. remove leading + trailing punctuation only: `\[](){}&|\'\"<>#:;~_^=\\-‑+*/\\\‘’“”，。《》…—‐•\`
3. then checking each token against the keyword

So far I can't really think of a use case that has leading and trailing punctuations, but to be safe, we could also provide an `exact` prop that skips step 2.

What do you think?

Reply:
This is not easy to do -- not sure if we should even try. Validation of text questions is not going to be that useful for most cases. Given the risk of false negatives (as well as false positives), as an author I'm likely to just give the answer in the explanation and let the reader figure out. 
Something related: It can be misleading to mark answer as correct when no keywords are specified. The reader might think the answer was validated when it was not.

Reply:
> I'm likely to just give the answer in the explanation and let the reader figure out.

I suspect this would be the majority use case as well, for questions that have long answers. Hence it's somewhat ok for the matching to be naive.

I was thinking more of mathematical / numerical questions with a single definite text answer, which the current algorithm may break (imagine a `keywords="1.99"` and the answer provided is `1.999` -- still correct by substring matching). Using the `1,2,3` above would fix this, but also makse the checking for long answers slightly stricter.

I'm not sure if we should advertise `text` questions as supporting mathematical / numerical questions as such, or defer it entirely to another pr and just use substring matching here 😮 

> It can be misleading to mark answer as correct when no keywords are specified. The reader might think the answer was validated when it was not.

fixed. it will still count toward the score in quizzes though.

Reply:
> I suspect this would be the majority use case as well, for questions that have long answers. Hence it's somewhat ok for the matching to be naive.

Let's add a note in the documentation to explain for which kind of cases the validation works and recommend to use it only for such cases. We don't want to give false hopes.

Reply:
Should we still at least perform tokenization + punctuation removal though? Since if that's the case, I think there's no harm in improving the algorithm to suit the majority of answers (which I think we can assume to be well formed) for text question validation.

Reply:
@damithc

Reply:
I'm not quite sure.  'Works for most cases' isn't very reassuring. We need to specify precisely the class of answers that works 100% and inform users to use the feature only for that class. We can't expect authors to experiment trial-and-error to find out whether to use the feature for each question.

For example, something like the following is safe:
> The validation works only if the expected answer is a precise phrase e.g., an equation (case insensitive, white space variations ignored)

Anything else that works 100%? If there is, we can add it to the above.

Supporting percentage matches is problematic because `not rewriting` and `rewriting` can both match the keyword `rewriting`.

Reply:
> The validation works only if the expected answer is a precise phrase e.g., an equation (case insensitive, white space variations ignored)

this dosen't work actually, in either algorithm, it's not the intended case

> I'm not sure if we should advertise text questions as supporting mathematical / numerical questions as such, or defer it entirely to another pr and just use substring matching here 😮

I think we should defer those types of questions to another type of question, with a smaller input box.

I intend for the matching to be only correct 80-90% of the time. It's difficult by nature to match this 100% like you mentioned, and to reach 99% of matches we likely need some other complex language processing which is out of scope. There's also the `show` button as a fallback so readers can always view the answer.

I was wondering if a simple substring match that matches potentially more things but also risks more false positives (`xx` -> `yxxy`) would be better or a slightly more restrictive one with tokenization and punctuation removal as mentioned above as such would be better in achieving that 80-90%, but,
having just realised there's also the issue of non whitespace tokenized languages 😱, I think we could just stick with the former.

Reply:
to further clarify I intend to add something like that to the docs:

> Warning: Keywords are validated by simply looking for the keyword as a pattern in the user's answer!
> This works well for some cases but not all. (and then a popover giving one good and one bad example)

Reply:
Can go ahead.